boards/arm/v2m_musca/v2m_musca.dts

```diff
@@ -0,0 +1,36 @@
+/dts-v1/;
+
+#include <arm/armv8-m.dtsi>
+
+#if defined (CONFIG_ARM_NONSECURE_FIRMWARE)
+#include "v2m_musca-nonsecure.dtsi"
```

Comment:
General comment here, about naming.
"<SOC name>-secure.dtsi is not so good, IMO, because I would like to preserve backwards compatibility with images without TrustZone consideration. So I would prefer that the "secure" .dtsi is the default .dtsi and, therefore, it should skip the "secure" from the name. And we should follow this convention for all ARMv8-M MCUs.

For Non-Secure: I would personally prefer to introduce a "_ns" suffix as a convention, to name the Non-Secure .dtsi headers. Unless you have strong preference on "-nonsecure" suffix, of course :)